src/util/file-filter.js

```diff
@@ -0,0 +1,109 @@
+/* @flow */
+import path from 'path';
+
+import minimatch from 'minimatch';
+
+import {createLogger} from './logger';
+
+const log = createLogger(__filename);
+
+// Use this function to mimic path.resolve without resolving to absolute path.
+export const normalizeResolve = (file: string): string => {
```

Comment:
https://github.com/eight04/web-ext/blob/71e3da42208cbca3fc98296c2e16c2a18f82d6fb/src/util/file-filter.js#L65
I want to make sure the result returned from `FileFilter.resolve` is consistent. Currently `fileFilter.resolve` has two behaviors. 

1. If `sourceDir` is set, all relative paths will be converted into absolute path by `path.resolve(sourceDir, file)`, which can represent as `path.join -> path.normalize -> trim trailing slash`.
2. Otherwise, just do `path.normalize -> trim trailing slash` to the path.

This make `sourceDir` optional and `fileFilter` can be run in "relative mode", which is the original behavior in `test.build.js`.